python/argo/interaction.py

```diff
@@ -106,31 +134,46 @@ class Command(Interaction):
     corresponding command's appropriate representation.
     """
 
-    def _result_and_state(self) -> Tuple[Any, Any]:
+    def _result_and_state_and_out_err(self) -> Tuple[Any, Any, str, str]:
```

Comment:
I think that once a three-tuple is exceeded, it might be a good idea to make a little record type to hold it. The `dataclass` decorator makes it not too verbose in Python.